tests/test_add_fund_and_claim.cairo

```diff
@@ -0,0 +1,418 @@
+use array::{Array, ArrayTrait, SpanTrait};
```

Comment:
I'd add tests for:
- the function `add_fund_to_salary_pools` to check the behavior if the `amounts` and `guilds` objects have different lengths of values (should revert)
- the function `update_cum_salary` that doesn't update the cum salary if it is already up-to-date?

Reply:
added `test_add_fund_length_mismatch`

Reply:
for verifying that `cum_salary ` is not updated, there are two ways to check, either verify that CumulativeSalaryUpdated is not emitted or check the gas used by function call, it should be much lesser (probably constant) in case when cum salary is not updated